App_Use_Cases.tex

```diff
@@ -8,4 +8,220 @@ \chapter{Use-Cases}
+
+The \refapi{PMIx_Fence} operation should have an option for the end user to specify which mode they desire for this operation.
+
+Additional information about the current state of the job (e.g., number of processes globally and locally) and of how the process was started (e.g., process binding) are also helpful. This "job level" information must be available immediately after PMIx_Init without the need for any explicit synchronization.
+
+The number of processes globally in the namespace and this process's rank within that namespace is important to know before establishing the Business Card information to best allocate resources.
+
+The number of processes local to the node and this process's local rank is important to know before establishing the Business Card information to help the caller determine the scope of the put operation. For example, to designate a leader to set up a shared memory segment of the proper size before putting that information into the locally scoped Business Card information.
+
+The number of processes local to a remote node is also helpful to know before establishing the Business Card information. This information is useful to pre-establish local resources before that remote node starts to initiate a connection or to determine the number of connections that need to be advertised in the Business Card when it is sent out.
+
+Note that some of the job level information may change over the course of the job in a dynamic application.
+
+\littleheader{Related Interfaces}
+
+{\large \refapi{PMIx_Put}}
+\pasteSignature{PMIx_Put}
+
+{\large \refapi{PMIx_Get}}
+\pasteSignature{PMIx_Get}
+
+{\large \refapi{PMIx_Commit}}
+\pasteSignature{PMIx_Commit}
+
+{\large \refapi{PMIx_Fence}}
+\pasteSignature{PMIx_Fence}
+
+{\large \refapi{PMIx_Init}}
+\pasteSignature{PMIx_Init}
+
+\littleheader{Related Keys}
+
+The following job level information is useful to have before establishing Business Card information:
+
+\pasteAttributeItem{PMIX_NODE_LIST}
+\pasteAttributeItem{PMIX_NUM_NODES}
+\pasteAttributeItem{PMIX_NODEID}
+\pasteAttributeItem{PMIX_JOB_SIZE}
+\pasteAttributeItem{PMIX_PROC_MAP}
+\pasteAttributeItem{PMIX_LOCAL_PEERS}
+\pasteAttributeItem{PMIX_LOCAL_SIZE}
+
+For each process this information is also useful (note that any one process may want to access this list of information about any other process in the system):
+
+\pasteAttributeItem{PMIX_RANK}
+\pasteAttributeItem{PMIX_LOCAL_RANK}
+\pasteAttributeItem{PMIX_GLOBAL_RANK}
+\pasteAttributeItem{PMIX_LOCALITY_STRING}
+\pasteAttributeItem{PMIX_HOSTNAME}
+
+There are other keys that are helpful to have before a synchronization point, this is not meant to be a comprehensive list.
+
+\section{Hybrid Programming Models}
```

Comment:
This commit is borked.  Rebase with the proper business card exchange commit and drop the hybrid programming model content.